API/Services/CacheService.cs

```diff
@@ -45,6 +45,8 @@ public interface ICacheService
 }
 public class CacheService : ICacheService
 {
+    private static readonly object syncLock = new();
```

Comment:
This should be named SyncLock. 

I think it might be better to use a thread safe lock rather than just an object, like I did in TagManagerService. Because each api call might end up being on a separate thread.

Reply:
I changed the lock. Now I'm using a SemaphoreSlim.